### PR TITLE
XIONE-12203 : Reimplement ScreenCapture using DRM for Realtek

### PR DESCRIPTION
**Pull request by maruku**
Reason for change:
  The original ScreenCapture plug-in implementation
  in rdkservices for Realtek platform is using vncframebuffer api
  (which is original implemented for Co-pilot VNC)
  Since https://ccp.sys.comcast.net/browse/RDK-37314,
  the api was removed. The getScreenShot() will fail in Realtek Xione
  platform.

  The implementation use Linux DRM to capture the screen.

Note: Only UI (without video) can be captured due to SVP is enabled.

Test Procedure: After device booting into UI, run the following console commands:
1. Activate the ScreenCapture plugin curl -s -d '{"jsonrpc":"2.0","id":"3","method": "Controller.1.activate", "params":{"callsign":"org.rdk.ScreenCapture"}}' http://127.0.0.1:9998/jsonrpc
2. Do uploadScreenCapture curl -s -d '{"method":"org.rdk.ScreenCapture.1.uploadScreenCapture","id":"3","jsonrpc":"2.0","params":{"url":"https://thunderstrikes.stb.r53.xcal.tv:9443/ce=ScreenshotO7c.png"}}' http://127.0.0.1:9998/jsonrpc There should be upload logs in /opt/logs/wpeframework.log

Priority: P1
Risks: low

Change-Id: I8923e89980421c838b79b2776e39438f7389f83d

### CODE REVIEW

**Comment by maruku** [comment]
Custom build for verify: https://jenkins.ccp.xcal.tv/jenkins/job/SKYXIONE-Yocto-Build/119040/

**Comment by maruku** [comment]
Hi @sborushevsky, 
 Since the original framebuffer api (for RealVNC)[ was removed](https://ccp.sys.comcast.net/browse/RDK-37314), I implement this ScreenCapture API using Linux DRM again.

**Comment by maruku** [comment]
Gerrit Build Verify Result for 26eb1c8
`
Patch Set 7: Verified+1

Primary builds completed:

    https://rdkportal.ccp.xcal.tv/builds/job/SERXHB1-Yocto-Build/17370 - default - SUCCESS - GRT size: 71.72 MB, CI size: 71.72 MB, difference:0 Bytes, oversized: false
    https://rdkportal.ccp.xcal.tv/builds/job/ARRISXG1V4-Yocto-Build/262036 - default - SUCCESS - GRT size: 115.65 MB, CI size: 115.66 MB, difference:-12272 Bytes, oversized: false
    https://rdkportal.ccp.xcal.tv/builds/job/RaspberryPi-CTH-Yocto-Build/55445 - default - SUCCESS
    https://rdkportal.ccp.xcal.tv/builds/job/ARRISXI6-Yocto-Build/241227 - default - SUCCESS - GRT size: 115.89 MB, CI size: 115.91 MB, difference:-20464 Bytes, oversized: false
    https://rdkportal.ccp.xcal.tv/builds/job/SKYXIONE-SKY-Yocto-Build/108467 - default - SUCCESS - GRT size: 272.28 MB, CI size: 272.28 MB, difference:4.00 KB, oversized: false
    https://rdkportal.ccp.xcal.tv/builds/job/TCHXi6-Yocto-Build/162474 - default - SUCCESS - GRT size: 122.55 MB, CI size: 122.56 MB, difference:-8176 Bytes, oversized: false
    https://rdkportal.ccp.xcal.tv/builds/job/SERCOMMXIONE-Yocto-Build/52009 - default - SUCCESS
    https://rdkportal.ccp.xcal.tv/builds/job/SERCOMMXIONE-US-FLEX2-Yocto-Build/7951 - default - SUCCESS
    https://rdkportal.ccp.xcal.tv/builds/job/ELEMENT-TEONE-Yocto-Build/45380 - default - SUCCESS
    https://rdkportal.ccp.xcal.tv/builds/job/PLATCO-CAD15-Yocto-Build/75483 - default - SUCCESS
    https://rdkportal.ccp.xcal.tv/builds/job/Pioneer-UHD-Yocto-Build/336 - default - SUCCESS
    https://rdkportal.ccp.xcal.tv/builds/job/HISENSE-A6GP-Yocto-Build/101894 - default - SUCCESS
    https://rdkportal.ccp.xcal.tv/builds/job/PLATCO-CAD11-Yocto-Build/98306 - default - SUCCESS
    https://rdkportal.ccp.xcal.tv/builds/job/HISENSE-V2-Yocto-Build/14304 - default - SUCCESS
    https://rdkportal.ccp.xcal.tv/builds/job/SKY-LLAMA-PANEL-Yocto-Build/119583 - default - SUCCESS
    [HISENSE-50](https://rdkportal.ccp.xcal.tv/builds/job/%3Ca%20href=)A6GX-Yocto-Build/50793" target="_blank" rel="nofollow">https://rdkportal.ccp.xcal.tv/builds/job/[HISENSE-50](https://ccp.sys.comcast.net/browse/HISENSE-50)A6GX-Yocto-Build/50793 - default - SUCCESS
    https://rdkportal.ccp.xcal.tv/builds/job/SKYXIONE-Yocto-Build/119087 - default - SUCCESS - GRT size: 240.06 MB, CI size: 240.06 MB, difference:0 Bytes, oversized: false
`

Jenkins Custom Build: 
* https://jenkins.ccp.xcal.tv/jenkins/job/SKYXIONE-Yocto-Build/119085/

**Comment by maruku** [comment]
Build verify done. 
https://gerrit.teamccp.com/#/c/739881/

`
Primary builds completed:

    https://rdkportal.ccp.xcal.tv/builds/job/SERXHB1-Yocto-Build/17370 - default - SUCCESS - GRT size: 71.72 MB, CI size: 71.72 MB, difference:0 Bytes, oversized: false
    https://rdkportal.ccp.xcal.tv/builds/job/ARRISXG1V4-Yocto-Build/262036 - default - SUCCESS - GRT size: 115.65 MB, CI size: 115.66 MB, difference:-12272 Bytes, oversized: false
    https://rdkportal.ccp.xcal.tv/builds/job/RaspberryPi-CTH-Yocto-Build/55445 - default - SUCCESS
    https://rdkportal.ccp.xcal.tv/builds/job/ARRISXI6-Yocto-Build/241227 - default - SUCCESS - GRT size: 115.89 MB, CI size: 115.91 MB, difference:-20464 Bytes, oversized: false
    https://rdkportal.ccp.xcal.tv/builds/job/SKYXIONE-SKY-Yocto-Build/108467 - default - SUCCESS - GRT size: 272.28 MB, CI size: 272.28 MB, difference:4.00 KB, oversized: false
    https://rdkportal.ccp.xcal.tv/builds/job/TCHXi6-Yocto-Build/162474 - default - SUCCESS - GRT size: 122.55 MB, CI size: 122.56 MB, difference:-8176 Bytes, oversized: false
    https://rdkportal.ccp.xcal.tv/builds/job/SERCOMMXIONE-Yocto-Build/52009 - default - SUCCESS
    https://rdkportal.ccp.xcal.tv/builds/job/SERCOMMXIONE-US-FLEX2-Yocto-Build/7951 - default - SUCCESS
    https://rdkportal.ccp.xcal.tv/builds/job/ELEMENT-TEONE-Yocto-Build/45380 - default - SUCCESS
    https://rdkportal.ccp.xcal.tv/builds/job/PLATCO-CAD15-Yocto-Build/75483 - default - SUCCESS
    https://rdkportal.ccp.xcal.tv/builds/job/Pioneer-UHD-Yocto-Build/336 - default - SUCCESS
    https://rdkportal.ccp.xcal.tv/builds/job/HISENSE-A6GP-Yocto-Build/101894 - default - SUCCESS
    https://rdkportal.ccp.xcal.tv/builds/job/PLATCO-CAD11-Yocto-Build/98306 - default - SUCCESS
    https://rdkportal.ccp.xcal.tv/builds/job/HISENSE-V2-Yocto-Build/14304 - default - SUCCESS
    https://rdkportal.ccp.xcal.tv/builds/job/SKY-LLAMA-PANEL-Yocto-Build/119583 - default - SUCCESS
    [HISENSE-50](https://rdkportal.ccp.xcal.tv/builds/job/%3Ca%20href=)A6GX-Yocto-Build/50793" target="_blank" rel="nofollow">https://rdkportal.ccp.xcal.tv/builds/job/[HISENSE-50](https://ccp.sys.comcast.net/browse/HISENSE-50)A6GX-Yocto-Build/50793 - default - SUCCESS
    https://rdkportal.ccp.xcal.tv/builds/job/SKYXIONE-Yocto-Build/119087 - default - SUCCESS - GRT size: 240.06 MB, CI size: 240.06 MB, difference:0 Bytes, oversized: false
`

**Review by anand-ky — CHANGES_REQUESTED** [review]
ScreenCapture plugin changes look good. Can you pls check and update unit tests.

File: Tests/CMakeLists.txt [hunk 1]
```diff
@@ -118,7 +116,6 @@ target_link_libraries(${PROJECT_NAME}
         ${NAMESPACE}DeviceIdentification
         ${NAMESPACE}FrameRate
         ${NAMESPACE}AVInput
-        ${NAMESPACE}ScreenCapture
```
[the remark this answers is not in the document]
Comment (maruku) [reply Tests/CMakeLists.txt]: Added back the ScreenCapture unit test

**Review by anand-ky — APPROVED** [review]

**Pull request merged**